Lib/http/__init__.py

```diff
@@ -149,3 +149,30 @@ def __new__(cls, value, phrase, description=''):
     NETWORK_AUTHENTICATION_REQUIRED = (511,
         'Network Authentication Required',
         'The client needs to authenticate to gain network access')
+
+
+@_simple_enum(StrEnum)
+class HTTPMethod:
+    """HTTP methods and descriptions
+
+    Methods from the following RFCs are all observed:
+
+        * RFC 7231: Hypertext Transfer Protocol (HTTP/1.1), obsoletes 2616
+        * RFC 5789: PATCH Method for HTTP
+    """
+    def __new__(cls, value, description):
+        obj = str.__new__(cls, value)
+        obj._value_ = value
+
+        obj.description = description
+        return obj
+
+    GET = 'GET', 'Transfer a current representation of the target resource.'
```

Comment:
These should be in some sort of order, probably alphabetical, possibly grouped by safety: `GET`, `HEAD`, `OPTIONS` / `DELETE`, `PATCH`, `POST`, `PUT` / `CONNECT`, `TRACE`.

Reply:
I used the same order from the RFCs, this is the usual order they are sorted (e.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)